packages/components/src/base_tooltip/index.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useRef, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { createPopper } from '@popperjs/core';
+import { getPopperOptions } from './utils';
+
+const BaseTooltip = (props) => {
+  const {
+    position,
+    bubble,
+    children,
+    tooltipClassName,
+    arrowClassName,
+    contentClassName,
+    ...cleanProps
+  } = props;
+
+  const content = useRef(null);
+  const tooltip = useRef(null);
+  const instance = useRef(null);
+  const arrow = useRef(null);
+
+  const mountPopper = useCallback(() => {
+    instance.current = createPopper(
+      content.current,
+      tooltip.current,
+      getPopperOptions(arrow.current, position),
+    );
+  }, [instance, content, tooltip, arrow, position]);
+
+  const destroyPopper = useCallback(() => {
+    if (instance.current) instance.current.destroy();
+  }, [instance]);
+
+  useEffect(() => {
+    mountPopper();
+    return () => {
+      destroyPopper();
+    };
+  }, [children]);
```

Comment:
Just inline `mountPopper` and `destroyPopper`, you may not even need the instance ref (you can keep it inside the effect scope)

Reply:
Please check if `children` in deps is really needed - we may not need to re-init popper on every re-render.